src/domains/transaction/components/TransactionDetail/TransactionDetails/TransactionDetails.tsx

```diff
@@ -17,13 +17,22 @@ export const TransactionDetails = ({
 	const format = useTimeFormat();
 
 	const timestamp = transaction.timestamp();
-	const data = transaction.data().data;
-
 	const { blockHeight } = useBlockHeight({
 		blockId: transaction.blockId(),
 		network: transaction.wallet().network(),
 	});
 
+	const nonce = useCallback(() => {
+		try {
+			const data = transaction.data().data;
+			const nonceValue = typeof data === "function" ? transaction.data().data().nonce : data?.nonce;
+
+			return typeof nonceValue === "string" ? nonceValue : "";
```

Comment:
Exactly, there are 3 implementations of this component:
- Recent transactions that are untyped, where the `nonce` is just a string.
- Transaction history, where the same modal is invoked but `nonce` comes inside `data().data()`.
- And then there's a 3rd one for multi-signature transactions that can return an empty object (that's why I am checking the type is a string before returning the `""` for `N/A`. 

Since that component isn't typed, passing a `nonce` prop would be a better approach. I would also think of a custom hook in case we have a similar issue with the block id.

Reply:
ok let's keep it as is for now until we make an adjustment to the PSDK